src/app/(proper_react)/(redesign)/(authenticated)/admin/churn-subscribers/ChurnAdmin.tsx

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { useEffect, useState } from "react";
+import { useSession } from "next-auth/react";
+import styles from "./ChurnAdmin.module.scss";
+import { SubscriberChurnRow } from "knex/types/tables";
+import { getAllChurns, upsertAllChurns } from "./actions";
+
+export const ChurnAdmin = () => {
+  const session = useSession();
+  const [subscriberData, setSubscriberData] = useState<
+    SubscriberChurnRow[] | null
+  >(null);
+
+  // Load data on component mount
+  useEffect(() => {
+    const loadSubscriberData = async () => {
+      const data = await getAllChurns();
+      setSubscriberData(data);
+    };
+
+    void loadSubscriberData();
```

Comment:
Hmm, isn't it better to avoid loading the existing subscriber data into the page? It's not super-sensitive I don't think, but just to minimise who has access to what data.

In any case, there's no need to load it in an effect, if you do want to keep it. You can load it on the server side in `page.tsx` and then pass it to this component as a prop.

Reply:
I don't think it matters in this case since the data is uploaded by the same person. It's just a way to sanity check what's updated in the db and make sure nothing weird happened

Reply:
Oh, I thought it also showed what was already in the database before the person uploaded anything? And this page is visible to _any_ admin, right?

Reply:
correct